src/util/sandbox/fd_sandbox.c

```diff
@@ -633,8 +634,10 @@ fd_sandbox_private_enter_no_seccomp( uint        desired_uid,
 
   /* Unshare everything in the parent user namespace, so that the nested
      user namespace does not have privileges over them. */
-  if( -1==unshare( CLONE_NEWNS | CLONE_NEWNET | CLONE_NEWCGROUP | CLONE_NEWIPC | CLONE_NEWUTS ) )
-    FD_LOG_ERR(( "unshare(CLONE_NEWNS | CLONE_NEWNET | CLONE_NEWCGROUP | CLONE_NEWIPC | CLONE_NEWUTS) failed (%i-%s)", errno, fd_io_strerror( errno ) ));
+  int flags = CLONE_NEWNS | CLONE_NEWCGROUP | CLONE_NEWIPC | CLONE_NEWUTS;
+  if( !keep_host_networking ) flags |= CLONE_NEWNET;
+
+  if( -1==unshare( flags ) ) FD_LOG_ERR(( "unshare(CLONE_NEWNS | CLONE_NEWNET | CLONE_NEWCGROUP | CLONE_NEWIPC | CLONE_NEWUTS) failed (%i-%s)", errno, fd_io_strerror( errno ) ));
```

Comment:
The log message will be a little inaccurate in this case, but it's not a big deal.